texar/core/attention_mechanism.py

```diff
@@ -846,46 +757,43 @@ def forward(self,  # type: ignore
         r"""Score the query based on the keys and values.
 
         Args:
-            query: Tensor of dtype matching `self.values` and shape
-                `[batch_size, query_depth]`.
-            state: Tensor of dtype matching `self.values` and shape
-                `[batch_size, alignments_size]`
-                (`alignments_size` is memory's `max_time`).
-            memory: The memory to query; usually the output of an RNN encoder.
-                This tensor should be shaped `[batch_size, max_time, ...]`.
-            memory_sequence_length: (optional) Sequence lengths for the batch
+            query: tensor, shaped ``[batch_size, query_depth]``.
+            state: tensor, shaped ``[batch_size, alignments_size]``
+                (``alignments_size`` is memory's ``max_time``).
+            memory: the memory to query; usually the output of an RNN encoder.
+                This tensor should be shaped ``[batch_size, max_time, ...]``.
+            memory_sequence_length (optional): sequence lengths for the batch
                 entries in memory.  If provided, the memory tensor rows are
                 masked with zeros for values past the respective sequence
                 lengths.
 
         Returns:
-            Tensor of dtype matching `self.values` and shape
-            `[batch_size, alignments_size]` (`alignments_size` is memory's
-            `max_time`).
+            Tensor of dtype matching ``memory`` and shape
+            ``[batch_size, alignments_size]`` (``alignments_size`` is memory's
+            ``max_time``).
         """
         processed_query = self.query_layer(query) if self.query_layer else query
```

Comment:
Since this part is almost the same for the 4 types of attention mechanisms, consider moving it to the base class `forward` method so we can share code.